### DISCUSSION TITLE
Does live custom resolution work with webrtc ?

### DISCUSSION THREAD

**Discussion by alez007** [blakeblackshear/frigate#5353] [Ask A Question] [answered]
Hi, I'm doing detection at 2560x1920 and configured live height to 1080. I'm using 0.12 beta 7 and go2rtc to record and detect on the main stream. 

On home assistant, if I use jsmpeg on lovelace card, I can see image being 1440x1080 as expected but if I switch to webrtc, the image's resolution is the detect resolution (2560x1920). 

Is this behaviour expected ?

Thanks

**Comment by NickM-27** [comment, marked as answer]
Yes this is expected, changing the size would require transcoding the stream.

**Reply by alez007** [reply]
so what would be the best solution? reducing detect resolution to maybe 1080p ? I'm thinking keeping that high resolution will make the home assistant mobile app unusable when live viewing

**Reply by NickM-27** [reply]
WebRTC doesn't use detect so that won't help either. Either use jsmpeg for home assistant or have HA use WebRTC with the sub stream instead of the main streams

**Reply by quenthal** [reply]
> WebRTC doesn't use detect so that won't help either. Either use jsmpeg for home assistant or have HA use WebRTC with the sub stream instead of the main streams

What is the fastest way to add substream cameras from Frigate to HA?

**Reply by NickM-27** [reply]
> > WebRTC doesn't use detect so that won't help either. Either use jsmpeg for home assistant or have HA use WebRTC with the sub stream instead of the main streams
> 
> 
> 
> What is the fastest way to add substream cameras from Frigate to HA?

Set them up in go2rtc config inside frigate and then use WebRTC card v3

**Reply by NickM-27** [reply]
That's the fastest way, there are of course other ways

**Reply by quenthal** [reply]
> That's the fastest way, there are of course other ways

Thanks! Always looking for the most elegant way (without duplicate addons etc), not necessarely the fastest as I originally stated :)

**Reply by NickM-27** [reply]
To be honest that's still the way I'd do it, WebRTC card works much better than rtsp2webrtc in my experience
